dig.go

```diff
@@ -74,10 +81,27 @@ type Container struct {
 
 // New constructs a Container.
 func New(opts ...Option) *Container {
-	return &Container{
-		nodes: make(map[key]*node),
-		cache: make(map[key]reflect.Value),
+	c := &Container{
+		nodes:      make(map[key]*node),
+		cache:      make(map[key]reflect.Value),
+		wrapErrors: true,
 	}
+
+	for _, opt := range opts {
+		opt.applyOption(c)
+	}
+	return c
+}
+
+// WrapErrors enables or disables wrapping of errors returned by user
+// functions.
+//
+// If enabled, errors returned by user-provided constructors will be wrapped
+// with more contextual information to aid in debugging the failure.
+//
+// Defaults to true.
```

Comment:
If it's true by default and the only thing to do is turn it off, maybe we can have an unnamed option? `dig.DisableErrorWrap()` or something like that.

Reply:
dig.RootCause makes this moot but the reason was that `dig.WrapErrors(true/false)` is pretty self-documenting.